server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -1005,4 +1005,31 @@ public void test_index_nested_array() throws Exception {
             {"x":10,"y":[[1,2],[3,4]]}"""
         );
     }
+
+    @Test
+    public void test_generated_column_can_refer_to_a_non_string_partitioned_by_column() throws Exception {
+        String partition = new PartitionName(new RelationName("doc", "tbl"), List.of("2")).asIndexName();
+        SQLExecutor executor = SQLExecutor.builder(clusterService)
+            .addPartitionedTable("create table tbl(a int, parted int, gen_from_parted int as parted + 1) partitioned by (parted)")
+            .build();
+        DocTableInfo table = executor.resolveTableInfo("tbl");
+        Indexer indexer = new Indexer(
+            partition,
+            table,
+            new CoordinatorTxnCtx(executor.getSessionSettings()),
+            executor.nodeCtx,
+            column -> NumberFieldMapper.FIELD_TYPE,
+            List.of(
+                table.getReference(new ColumnIdent("a")),
+                table.getReference(new ColumnIdent("parted"))
```

Comment:
Unfortunately, this test passes even without applying patch.

In real execution we use `ColumnIndexWriterProjection` which doesn't include parted column into target columns and Indexers falls to [another code path](https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/execution/dml/Indexer.java#L171-L173) which doesn't have casting issue.

I tried to exclude "parted" column here in the test setup but then it fails with `"Number of values must be less than or equal the number of targetColumns/valueIndexers"`.

I would say let's drop this test as it's useless and keep integration test?